cime_config/config_grids.xml

```diff
@@ -5303,8 +5313,9 @@
     </gridmap>
 
     <gridmap glc_grid="mpas.gis20km" ocn_grid="IcoswISC30E3r5">
-      <map name="OCN2GLC_FMAPNAME">cpl/gridmaps/IcoswISC30E3r5/map_IcoswISC30E3r5_to_gis20km_esmfaave.20240403.nc</map>
-      <map name="OCN2GLC_SMAPNAME">cpl/gridmaps/IcoswISC30E3r5/map_IcoswISC30E3r5_to_gis20km_esmfbilin.20240403.nc</map>
+      <map name="OCN2GLC_SHELF_FMAPNAME">cpl/gridmaps/IcoswISC30E3r5/map_IcoswISC30E3r5_to_gis20km_esmfaave.20240403.nc</map>
+      <map name="OCN2GLC_SHELF_SMAPNAME">cpl/gridmaps/IcoswISC30E3r5/map_IcoswISC30E3r5_to_gis20km_esmfbilin.20240403.nc</map>
+      <map name="OCN2GLC_TF_SMAPNAME">cpl/gridmaps/IcoswISC30E3r5/map_IcoswISC30E3r5_to_gis20km_esmfneareststod.20240422.deeperThan300m.nc</map>
```

Comment:
For now I have a standalone python file that looks like this:
```
#!/usr/bin/env python
"""
Script for defining mask of ocean cells that are deeper than a threshold.
To be used for creating a scrip file for remapping that only remaps
from ocean cells deeper than that threshold.
"""

import argparse
from netCDF4 import Dataset

parser = argparse.ArgumentParser(
                    description='mask out shallow ocean cells')
parser.add_argument("-s", dest="scrip",
                    required=True,
                    help="scrip file to add mask to")
parser.add_argument("-m", dest="mpas",
                    required=True,
                    help="MPAS-Ocean mesh file")
parser.add_argument("-d", dest="depth",
                    required=False, help="depth threshold (m)",
                    type=float, default=300.0)
args = parser.parse_args()

# open mesh file
fmesh = Dataset(args.mpas, 'r')
nCells = len(fmesh.dimensions['nCells'])
bottomDepth = fmesh.variables['bottomDepth'][:]
mask = bottomDepth > args.depth
fmesh.close()

fscrip = Dataset(args.scrip, 'r+')
if 'grid_imask' not in fscrip.variables:
    fscrip.createVariable('grid_imask', 'i', ('nCells',))
fscrip.variables['grid_imask'][:] = mask
fscrip.close()
```

The mask gets applied automatically by ESMF_RegridWeightGen, see 12.2.2 Masking section of docs:
https://earthsystemmodeling.org/docs/release/ESMF_8_0_1/ESMF_refdoc/node3.html#SECTION03022200000000000000

It may be possible to achieve a similar result with some NCO commands, but I suspect this short script is easier to understand and document.  If you agree, I think MPAS-Tools is the most logical place to put it for now, unless you have another suggestion.

Reply:
Yes, I agree, please add this to MPAS-Tools.

Reply:
Ok, sounds good.  I'll do that when I get back from the holiday.

Reply:
@xylar , I've added an MPAS-Tools PR with that script and instructions for how to use it in the workflow here: https://github.com/MPAS-Dev/MPAS-Tools/pull/578
I wasn't lying when I said I'd get to it after the holiday, it was just at _little_ later than implied. :)